packages/mui-material/src/Select/SelectInput.js

```diff
@@ -257,6 +269,56 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     }
   };
 
+  React.useEffect(() => {
+    if (isPointerDown) {
+      const doc = ownerDocument(displayRef.current);
+
+      const handleGlobalMouseUp = (event) => {
+        setIsPointerDown(false);
+
+        if (dragSelectRef.current.isDragging) {
+          // If we're dragging and the mouse is released, check where it was released
+          dragSelectRef.current.isDragging = false;
+
+          // Check if mouse is over a menu item
+          const targetElement = event.target;
+          let menuItem = null;
+
+          // Find if we released on a menu item (checking up the parent chain)
+          let current = targetElement;
+          while (current && !menuItem) {
+            // Check if this element has role="option"
+            if (current.getAttribute && current.getAttribute('role') === 'option') {
+              menuItem = current;
+            }
+            current = current.parentElement;
+          }
+
+          if (!menuItem) {
+            // If released outside menu items, close the menu
+            update(false, event);
+          }
+        }
+      };
+
+      const handleGlobalMouseMove = () => {
+        if (isPointerDown) {
+          dragSelectRef.current.isDragging = true;
+        }
+      };
```

Comment:
Is it necessary to check whether the pointer is dragging? Isn't it enough to check what the mouse up event target is?